src/components/Sidebar.tsx

```diff
@@ -190,61 +222,95 @@ const Sidebar = ({ onClose, ...rest }: SidebarProps) => {
+                          onClose={onClose} />)}
+          <ManagerDropdownMenu />
           {mentorshipItems?.length > 0 && <Divider marginY={2} />}
-
           {mentorshipItems.map(item => <SidebarRow key={item.path} item={item}
             onClose={onClose} />)}
         </Box>
-        <HStack spacing={{ base: '0', [sidebarBreakpoint]: '6' }} >
-          {/* <IconButton
-            size="lg"
-            variant="ghost"
-            aria-label="open menu"
-            icon={<FiBell />}
-            /> */}
-          <Menu>
-            <MenuButton
-              marginX={4}
-              marginBottom={4}
-              paddingLeft={4}
-              py={2}
-              transition="all 0.3s"
-              _focus={{ boxShadow: 'none' }}>
-              <HStack>
-                <Avatar
-                  size={'sm'}
-                  bg="brand.a"
-                  color="white"
-                  name={formatUserName(me.name)} />
-                <Text
-                  display={{ base: 'flex', [sidebarBreakpoint]: 'flex' }}
-                  fontSize="sm">
-                  {formatUserName(me.name)} </Text>
-                <Box display={{ base: 'flex', [sidebarBreakpoint]: 'flex' }}><FiChevronRight /></Box>
-              </HStack>
-            </MenuButton>
-            <MenuList
-              bg={useColorModeValue('white', 'gray.900')}
-              borderColor={useColorModeValue('gray.200', 'gray.700')}>
-              <MenuItem as={NextLink} href='/profile'>个人信息</MenuItem>
-              <MenuDivider />
-              <MenuItem as={NextLink} href='/who-can-see-my-data'>
-                <LockIcon marginRight={1} />谁能看到我的数据
-              </MenuItem>
-              <MenuDivider />
-              <MenuItem onClick={() => signOut()}>退出登录</MenuItem>
-            </MenuList>
-          </Menu>
-        </HStack>
+        <Box>
+          <UserDropdownMenu />
+        </Box>
       </Flex>
     </Box >
   );
 };
 
 export default Sidebar;
 
+function UserDropdownMenu() {
+  const [user] = useUserContext();
+  const name = formatUserName(user.name);
+  const backgroundColor = useColorModeValue('white', 'gray.900');
+  const boderColor = useColorModeValue('gray.200', 'gray.700'); 
+
+  return <Menu placement='right-start'>
+    <DropdownMenuButton title={name} 
+      component={<Avatar size={'sm'} bg="brand.a" color="white" name={name} />} />
+    <MenuList bg={backgroundColor} borderColor={boderColor}>
+      {userDropdownMenuItems.map(item =>
+        <>
+          <MenuItem 
+            as={typeof item.action === 'string' ? NextLink : undefined}
+            href={typeof item.action === 'string' ? item.action : undefined}
+            onClick={typeof item.action === 'function' ? item.action : undefined}
+            >
+              {item.icon}{item.name}
+          </MenuItem>
+        </>)}
+    </MenuList>
+  </Menu>;
+}
+
+function ManagerDropdownMenu() {
```

Comment:
why not merge ManagerDropdownMenu and UserDropdownMenu to one function `DropdownMenu`, which takes title, icon, and a DropdownMenuItem object as input parameters?

Reply:
<img width="1292" alt="Screenshot 2024-10-18 at 14 57 02" src="https://github.com/user-attachments/assets/08b723f0-c34e-4444-8569-62b77a689132">
<Avatar /> require name to be type: string | undefined
but user.name is type: string | null.